sources/packages/backend/apps/api/src/route-controllers/report/report.controller.service.ts

```diff
@@ -1,20 +1,85 @@
-import { Injectable } from "@nestjs/common";
-import { getFileNameAsCurrentTimestamp } from "@sims/utilities";
+import {
+  BadRequestException,
+  Injectable,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import {
+  ReportService,
+  FILTER_PARAMS_MISMATCH,
+  REPORT_CONFIG_NOT_FOUND,
+} from "@sims/services";
+import {
+  getFileNameAsCurrentTimestamp,
+  CustomNamedError,
+} from "@sims/utilities";
 import { Response } from "express";
 import { Readable } from "stream";
+import { FormService } from "../../services";
+import { MinistryReportsFilterAPIInDTO } from "./models/report.dto";
+import { FormNames } from "../../services/form/constants";
 
 /**
- * Service layer for reports.
+ * Controller Service layer for reports.
  */
 @Injectable()
 export class ReportControllerService {
+  constructor(
+    private readonly reportService: ReportService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Generates report in csv format.
+   * @param payload report filter payload.
+   * @param response http response as file.
+   * @param options related options.
+   * - `institutionId` related institution id.
+   */
+  async generateReport(
+    payload: MinistryReportsFilterAPIInDTO,
```

Comment:
We should use an interface that would accommodate Ministry and Institution since this is shared between both. Does it make sense?